src/xensiv_pas_gas_a2l.c

```diff
@@ -122,32 +122,56 @@ int32_t xensiv_pas_gas_a2l_get_gas_config(const xensiv_pas_gas_t *dev, xensiv_pa
 int32_t xensiv_pas_gas_a2l_set_alarm_hysteresis(const xensiv_pas_gas_t *dev, uint16_t alarm_hys) {
     xensiv_pas_gas_plat_assert(dev != NULL);
 
-    alarm_hys = (uint16_t)xensiv_pas_gas_plat_htons(alarm_hys);
-    return xensiv_pas_gas_set_reg(dev, (uint8_t)XENSIV_PAS_GAS_A2L_REG_ALARM_HYS_H, (uint8_t *)&alarm_hys, 2U);
+    // Only 15 bits are valid, mask input
+    alarm_hys &= 0x7FFF;
```

Comment:
Should not the xensiv_pas_gas_plat_htons() take care of this endianess?

Reply:
.In this context, the issue is not really about endianness, but rather the layout of the registers. We cannot write the 16-bit values as a whole because they are divided into separate registers with specific bits reserved. So we have to do the manual packing or ?